catalog/ui/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/metacard/ListApplication.java

```diff
@@ -0,0 +1,368 @@
+import org.apache.commons.io.IOUtils;
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.platform.util.uuidgenerator.UuidGenerator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.Response;
+import spark.servlet.SparkApplication;
+
+public class ListApplication implements SparkApplication {
+
+  private static final Logger LOGGER = LoggerFactory.getLogger(ListApplication.class);
+
+  private static final String LIST_TYPE_HEADER = "List-Type";
+
+  private static final String DEFAULT_FILE_EXTENSION = "bin";
+
+  private static final String DEFAULT_MIME_TYPE = "application/octet-stream";
+
+  private static final String DEFAULT_FILE_NAME = "file";
+
+  private static final List<String> REFINEABLE_MIME_TYPES =
+      Arrays.asList(DEFAULT_MIME_TYPE, "text/plain");
+
+  private final MimeTypeMapper mimeTypeMapper;
+
+  private final CatalogFramework catalogFramework;
+
+  private final UuidGenerator uuidGenerator;
+
+  private final SplitterLocator splitterLocator;
+
+  public ListApplication(
+      MimeTypeMapper mimeTypeMapper,
+      CatalogFramework catalogFramework,
+      UuidGenerator uuidGenerator,
+      SplitterLocator splitterLocator) {
+    this.mimeTypeMapper = mimeTypeMapper;
+    this.catalogFramework = catalogFramework;
+    this.uuidGenerator = uuidGenerator;
+    this.splitterLocator = splitterLocator;
+  }
+
+  @Override
+  public void init() {
+    post(
+        "/list/import",
+        (request, response) -> {
+          MultipartConfigElement multipartConfigElement = new MultipartConfigElement("");
+          request.raw().setAttribute("org.eclipse.jetty.multipartConfig", multipartConfigElement);
+
+          String listType = request.headers(LIST_TYPE_HEADER);
+
+          if (StringUtils.isBlank(listType)) {
+            String exceptionMessage = String.format("The header %s must be set.", LIST_TYPE_HEADER);
+            LOGGER.info(exceptionMessage);
+            createBadRequestResponse(exceptionMessage, response);
+            return null;
+          }
+
+          List<Part> parts = new ArrayList<>(request.raw().getParts());
+
+          CreateInfo createInfo = parseAttachments(parts);
+
+          if (createInfo == null) {
+            String exceptionMessage = "Unable to parse the attachments.";
+            LOGGER.debug(exceptionMessage);
+            createBadRequestResponse(exceptionMessage, response);
+            return null;
+          }
+
+          for (Splitter splitter : lookupSplitters(createInfo.contentType)) {
+            List<String> ids = new LinkedList<>();
+            List<String> errorMessages = new LinkedList<>();
+
+            boolean isSplitSuccessful = false;
+
+            try (Stream<StorableResource> stream = createStream(createInfo, splitter, listType)) {
+              isSplitSuccessful = true;
+              stream
+                  .sequential()
+                  .peek(storableResource -> appendMessageIfError(errorMessages, storableResource))
```

Comment:
❓ According to the javadoc of peek, it should only be used for debugging (and probably not for doing side effects). We could change this to be good citizens 🙇

Reply:
Changed to a `map`.